### PR TITLE
Congeal root index.js with CLI and extension

### PR DESCRIPTION
**Pull request by paulirish**
This is a followup to the large folder refactor PR (#243).

------------------

`/index.js` is currently only used for the CLI case, which is a bit confusing. This change makes `cli.js` completely in line with the extension's `background.js`.

##### Other changes:

* the big diff in cli/index.js (https://github.com/paulirish/lighthouse/pull/3/commits/b53d490d901f71e1ecb6c951c57b256c888594cc#diff-b6e52b89ad89c29a17bf6a53e65888f9L52) is mostly moving things around & formatting
* In the extension: `app.js` doesnt run the app, so its now renamed to `popup.js`
* More logging
* Small readme addition

### CODE REVIEW

File: scripts/netdep-graph.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+# Usage:
+#     CHROMIUM_SOURCE_PATH=$HOME/chromium/src ./netdep-graph.sh
+#     open network-dependency-graph.png
+
+if [[ x"$CHROMIUM_SOURCE_PATH" == x ]]; then
```
Comment (samccone) [inline scripts/netdep-graph.sh]: heh funny

File: extension/gulpfile.js [hunk 2]
```diff
@@ -98,7 +99,11 @@ gulp.task('browserify', () => {
     .pipe(gulp.dest('dist/scripts'));
 });
 
-gulp.task('clean', del.bind(null, ['.tmp', 'dist']));
+gulp.task('clean', () => {
+  del(['.tmp', 'dist', 'app/scripts']).then(paths =>
```
Comment (brendankenny) [inline extension/gulpfile.js]: I think you want a `return del(...)...` here so gulp can pipeline properly

File: cli/index.js [hunk 3]
```diff
@@ -40,37 +49,32 @@ const cli = meow(`
...
+    if (err.code === 'ECONNREFUSED') {
+      console.error('Unable to connect to Chrome. Did you run ./launch-chrome.sh?');
+    } else {
+      console.error('Runtime error encountered:', err);
+      console.error(err.stack);
+    }
+    process.exit(1);
+  });
 
+// set logging preferences
```
Comment (brendankenny) [inline cli/index.js]: logging level should probably be set above the `lighthouse()` call above. No logging in there currently, but if we ever add some, we'll want the logging level already set.

**Comment by brendankenny** [comment]
two small things but otherwise LGTM

**Comment by paulirish** [comment]
@brendankenny thx. both were really good suggestions. now addressed.

**Pull request merged**